File: MusicMaker/Core/Wave/square_wave.cpp
//
//  square_wave.cpp
//  MusicMaker
//
//  Created by Sergey Smagleev on 10.07.20.
//  Copyright © 2020 Sergey Smagleev. All rights reserved.
//

#include "square_wave.hpp"
#include "math.h"

float CSquareWave::shape(float value) {
    return fmod(value, M_PI * 2) < M_PI ? 1 : 0;
}

CSquareWave::CSquareWave(float _frequency, float _amplitude, float _phaseShift) : CWave(_frequency, _amplitude, _phaseShift) { }

CSquareWave::~CSquareWave() { }

File: MusicMaker/MusicEngine/SoundSynthesizer/Waves/SquareWave.swift
//
//  SquareWave.swift
//  MusicMaker
//
//  Created by Sergey Smagleev on 28.05.20.
//  Copyright © 2020 Sergey Smagleev. All rights reserved.
//

import Foundation

class SquareWave: Wave {
    
    override func shape(_ value: Float) -> Float {
        return Int(value / Float.pi) % 2 == 0 ? 1 : -1
    }
    
}
